### ISSUE TITLE
--deploy-url not working as expected

### ISSUE THREAD

**Issue by zackarychapple** [angular/angular-cli#8573]
### Versions
```
Angular CLI: 1.5.0
Node: 8.8.1
OS: darwin x64
Angular: 4.4.3
... animations, common, compiler, compiler-cli, core, forms
... http, language-service, platform-browser
... platform-browser-dynamic, router, tsc-wrapped

@angular/cli: 1.5.0
@angular/material: 2.0.0-beta.7
@angular-devkit/build-optimizer: 0.0.32
@angular-devkit/core: 0.0.20
@angular-devkit/schematics: 0.0.35
@ngtools/json-schema: 1.1.0
@ngtools/webpack: 1.8.0
@schematics/angular: 0.1.0
typescript-collections: 1.2.5
typescript: 2.4.2
webpack-bundle-analyzer: 2.9.0
webpack: 3.8.1
```

### Repro steps
* `ng build --deploy-url=http://ermygod.cdn/folder`
Inline assets work as expected from index.html, however lazy loaded assets do not load. 
Vendor bundle
`http://ermygod.cdn/folder/vendor.bundle.js`
Lazy Loaded chunks do not load, they are missing trailing slash.
`http://ermygod.cdn/folder0.chunk.js`
* `ng build --deploy-url=http://ermygod.cdn/folder/`
Both work as expected

### Desired behavior
The trailing slash and non trailing slash should behave the same.

**Comment by asadsahi** [comment]
Hi guys, I have just upgraded to cli v1.6.0-rc.0 and this command stops working:

`    "start": "ng serve --deploy-url=/dist/ --output-hashing=media --extract-css",`

Getting this error:

```
An asset cannot be written to a location outside the project.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! aspnetnetcore@0.0.0 start: `ng serve --deploy-url=/dist --output-hashing=media --extract-css`
npm ERR! Exit status 1
npm ERR!
```

**Comment by clydin** [comment]
Fixed via #8740
